### ISSUE TITLE
Weird bug in RoiAlign

### ISSUE THREAD

**Issue by Hyunseok-Kim0** [PINTO0309/onnx2tf#22]
### Issue Type

Others

### onnx2tf version number

1.1.33

### Download URL for ONNX

Please use code below to reproduce bug.

```python
import torch
from einops import rearrange
from torch import nn
import numpy as np
from onnx2tf import convert
import shutil
import torchvision
import tensorflow as tf
import cv2
import matplotlib.pyplot as plt
import requests
import onnxruntime


class dummy_network(nn.Module):

    def __init__(self, output_size, spatial_scale):
        super(dummy_network, self).__init__()

        self.output_size = output_size
        self.spatial_scale = spatial_scale

    def forward(self, x, roi):
        x = torchvision.ops.roi_align(x, boxes=roi, output_size=self.output_size, spatial_scale=self.spatial_scale)

        return x


def visualize(outputs):
    plt.figure(figsize=(8, 8))

    col = 4
    row = len(outputs) // 4 + 1

    for i, o in enumerate(outputs, start=1):
        plt.subplot(row, col, i)
        plt.axis("off")
        plt.imshow(o.astype(int))

    plt.show()


def main():
    model = dummy_network(output_size=(64, 64), spatial_scale=1.0)
    model.eval()

    dummy_name = "dummy_roi_align"
    onnx_save_path = f"tflite/{dummy_name}.onnx"
    temp_tflite = "tflite/model_float32.tflite"
    tflite_save_path = f"tflite/{dummy_name}.tflite"

    url = "https://static01.nyt.com/images/2021/09/14/science/07CAT-STRIPES/07CAT-STRIPES-mediumSquareAt3X-v2.jpg"
    image_nparray = np.asarray(bytearray(requests.get(url).content), dtype=np.uint8)
    image = cv2.imdecode(image_nparray, cv2.IMREAD_COLOR)

    dummy_input_x = np.expand_dims(cv2.resize(cv2.cvtColor(image, cv2.COLOR_BGR2RGB), (256, 256)), axis=0)
    dummy_input_x = rearrange(torch.Tensor(dummy_input_x), "n h w c -> n c h w")
    # dummy_input_roi = [[0, 0, 0, 0.5, 0.5], [0, 0.5, 0, 1, 0.5], [0, 0, 0.5, 0.5, 1], [0, 0.5, 0.5, 1, 1]]
    dummy_input_roi = [[0, 0, 0, 128, 128], [0, 128, 0, 256, 128], [0, 0, 128, 128, 256], [0, 128, 128, 256, 256]]
    # dummy_input_roi = [[0, 0, 1, 2, 3]]
    # dummy_input_roi = [[0, 0, 0, 256, 256]]
    dummy_input_roi = torch.Tensor(dummy_input_roi)

    torch.onnx.export(model,
                      args=(dummy_input_x, dummy_input_roi),
                      f=onnx_save_path,
                      input_names=["x", "roi"],
                      opset_version=11)

    convert(onnx_save_path, output_folder_path="tflite")
    shutil.move(temp_tflite, tflite_save_path)


    # get torch output
    # -----------------------------------------------------------------------------------------------
    with torch.no_grad():
        torch_output = model(dummy_input_x, dummy_input_roi)
        torch_output = rearrange(torch_output, "n c h w -> n h w c")

    # visualize(torch_output)

    # get onnx output
    # -----------------------------------------------------------------------------------------------
    onnx_session = onnxruntime.InferenceSession(onnx_save_path, providers=['CPUExecutionProvider'])
    onnx_inputs = dict(x=dummy_input_x.numpy(), roi=dummy_input_roi.numpy())
    onnx_output = onnx_session.run(None, onnx_inputs)[0]
    onnx_output = rearrange(onnx_output, "n c h w -> n h w c")

    # compare torch output and onnx output
    np.testing.assert_allclose(onnx_output, torch_output.numpy())

    # get tflite output
    # -----------------------------------------------------------------------------------------------
    tflite_onnx2tf = tf.lite.Interpreter(model_path=tflite_save_path)
    tflite_onnx2tf.allocate_tensors()
    tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[0]['index'],
                              rearrange(dummy_input_x.numpy(), "n c h w -> n h w c"))
    tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[1]['index'], dummy_input_roi.numpy())
    tflite_onnx2tf.invoke()
    tflite_output = [tflite_onnx2tf.get_tensor(i['index']) for i in tflite_onnx2tf.get_output_details()][0]

    total_output = np.concatenate([torch_output, onnx_output, tflite_output], axis=0)
    visualize(total_output)

    print("convert done")

    return


if __name__ == "__main__":
    main()
```

### Parameter Replacement JSON

```json
N/A
```

### Description

1. Purpose: Personal development

2. What: When I tested RoiAlign, it showed unexpected output as below. 

| pytorch output |
|:---------------:| 
|<img src="https://user-images.githubusercontent.com/34959032/203920865-733943c3-8c7a-4044-936e-b63b408e767b.png" width="560" height="140">|
| onnx output |
|<img src="https://user-images.githubusercontent.com/34959032/203920910-83cabf01-727c-49be-92d5-2fea5e846e24.png" width="560" height="140">|
| wrong tflite output |
|<img src="https://user-images.githubusercontent.com/34959032/203920938-f6896615-b85d-4a9f-b87f-051b9dad2191.png" width="560" height="140">|
| correct tflite output after modification |
|<img src="https://user-images.githubusercontent.com/34959032/203920997-c84b949c-e45c-43d5-8f41-8cd4f1dd3d34.png" width="560" height="140">|

3. How: After some debugging, I found that wrong order of roi coordinates are fed into tflite. After changing line 124 to `x0, x1, y1, y0 = tf.split(boxes, 4, axis=1)`, I could get correct result as shown above.
https://github.com/PINTO0309/onnx2tf/blob/0b14623068a7204e48c73eb996ac8e529b99b442/onnx2tf/ops/RoiAlign.py#L116-L126
Although the output value is slightly different as below due to the implementation detail in tensorflow, it looks working fine. However, I couldn't find out why order of the roi coordinates is changed from `x0, y0, x1, y1` to `x0, x1, y1, y0`. 
<img src="https://user-images.githubusercontent.com/34959032/203922607-64370e19-8924-4939-9b95-41a388f13401.png" width="400" height="250">

**Comment by PINTO0309** [comment]
I have also tried tracing, but so far I can't figure out why the coordinate values are inverted.

**Comment by PINTO0309** [comment]
I have read the TensorFlow tf.image.crop_and_resize spec and the C++ logic. 
https://www.tensorflow.org/api_docs/python/tf/image/crop_and_resize

It appeared to be a specification that internally performs a Flip on its own when the coordinate values are inverted. Therefore, try the following one line
https://github.com/PINTO0309/onnx2tf/blob/bdb6c35683274aaac8c42ffc1207bc85c74893b2/onnx2tf/ops/RoiAlign.py#L156

I added a process to invert the large/small relationship in advance.
```python
            y0 = tf.math.minimum(ny0, ny0 + nh)
            y1 = tf.math.maximum(ny0, ny0 + nh)
            x0 = tf.math.minimum(nx0, nx0 + nw)
            x1 = tf.math.maximum(nx0, nx0 + nw)
            return tf.concat([y0, x0, y1, x1], axis=1)
```
As a result, the vertical flipping bug appears to be fixed.
![image](https://user-images.githubusercontent.com/33194443/204085259-586e2a1c-0dbc-41d9-910c-148a5ab65261.png)

**Comment by PINTO0309** [comment]
- Before
![image](https://user-images.githubusercontent.com/33194443/204089247-0fbae296-0ba7-4e60-abe9-c91d5ef0eaff.png)

- After
![image](https://user-images.githubusercontent.com/33194443/204089733-9678df4e-e7d8-41a7-8215-1b911f58c008.png)
![image](https://user-images.githubusercontent.com/33194443/204089912-5e4ad558-e6a0-44d1-a974-2cbae542b337.png)
